test/plug/adapters/cowboy2/conn_test.exs

```diff
@@ -406,8 +406,31 @@ defmodule Plug.Adapters.Cowboy2.ConnTest do
 
     assert {:ok, "OK"} = :hackney.body(client)
     :hackney.close(client)
-  after
-    :ok = Plug.Adapters.Cowboy2.shutdown(__MODULE__.HTTPS)
+  end
+
+
+  def http2(conn) do
+    assert %{adapter: {Plug.Adapters.Cowboy2.Conn, %{version: version}}} = conn
+    send_resp(conn, 200, Atom.to_string(version))
+  end
+
+  test "http2" do
+    opts = [
+      cacertfile: @https_options[:certfile], server_name_indication: 'localhost', port: 8004
+    ]
+
+
+    {:ok, _} = Application.ensure_all_started(:kadabra)
+    {:ok, pid} = Kadabra.open('localhost', :https, opts)
+    Kadabra.get(pid, "/http2")
+    receive do
```

Comment:
This block can be replaced with:

`assert_receive({:end_stream, %Kadabra.Stream.Response{status: 200, body: "HTTP/2"}, 1_000)`